### PR TITLE
Priya Power's Mod1 Week2 IND CHALLENGE

### PR DESCRIPTION
**Pull request by priyajag72**
Completed Iteration 1
Completed Iteration 2
Completed Iteration 3
Comments for Iteration 4
DID NOT Complete Iteration 4

### CODE REVIEW

File: lib/product.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  def initialize(category, name, unit_price, quantity)
+    @category = category
+    @name = name
+    @unit_price = unit_price
+    @quantity = quantity.to_i
+    @is_hoarded = false
+  end
+
+  def total_price
+    quantity * unit_price
```
Comment (mikedao) [inline lib/product.rb:14]: good

File: lib/shopping_cart.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  attr_reader :name
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity
+    @products = []
+  end
+
+  def capacity
+    @capacity.gsub(/[abcdefghijklmnopqrstuvwxyz]/, '')
```
Comment (mikedao) [inline lib/shopping_cart.rb:12]: iiiinnntereeesssting. I love it.

File: lib/shopping_cart.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  end
+
+  def add_product(product)
+    @products << product
+  end
+
+  def details
+    details = {
+      name: @name,
+      capacity: capacity
```
Comment (mikedao) [inline lib/shopping_cart.rb:27]: good reuse of methods.

File: lib/shopping_cart.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      true
+    else
+      false
+    end
+  end
+
+  def products_by_category(category)
+    @products.find do |product|
+      if product.category == category
+        p product
```
Comment (mikedao) [inline lib/shopping_cart.rb:50]: We want it to return a collection of product objects, not print them to the screen.

File: lib/shopping_cart.rb (hunk 2 above)
Comment (priyajag72) [reply lib/shopping_cart.rb:12]: You are awesome. I did learn about reg ex after this and feel like I could rewrite this now!!!

File: lib/shopping_cart.rb (hunk 3 above)
Comment (priyajag72) [reply lib/shopping_cart.rb:27]: Yay!

File: lib/shopping_cart.rb (hunk 4 above)
Comment (priyajag72) [reply lib/shopping_cart.rb:50]: Ahhhh. I think I follow return methods and brainstorming through my iteration first now. I would be ready to reattempt!

**Pull request closed without merging**